framework/src/multiapps/MultiApp.C

```diff
@@ -482,6 +482,20 @@ MultiApp::restore()
   }
 }
 
+void
+MultiApp::restoreSubApps()
+{
+  for (unsigned int i = 0; i < _apps.size(); i++)
+  {
+    for (auto & sub_app :
+         _apps[i]->getExecutioner()->feProblem().getMultiAppWarehouse().getObjects())
```

Comment:
It's hilarious to me that the `MultiApp` needs to do this in order to get the sub-apps. This is beyond the scope of your PR though